---
title: "An√°lisis de datos de RNA-Seq - Splicing Alternativo"
subtitle: "BioScienceApps"
author: "Estefania Mancini"
date: "18-21 de abril de 2022"
output:
    html_document:
    toc: true # table of content true
    toc_depth: 3  # upto three depths of headings (specified by #, ## and ###)
    toc_float: true
    code_folding: hide
    fig_caption: yes
    fig_height: 6
    fig_width: 6
    number_sections: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# ASpli Example
```{r design, echo=TRUE, eval=TRUE}
library(ASpli)
library(GenomicFeatures)

# gtf preprocessing ----
gtfFileName <- aspliExampleGTF()
genomeTxDb  <- makeTxDbFromGFF( gtfFileName )

# feature extraction ----
features    <- binGenome( genomeTxDb )

#bams and target file ----
BAMFiles <- aspliExampleBamList()
targets  <- data.frame(row.names = paste0('Sample',c(1:6)),
                       bam = BAMFiles[1:6],
                       f1  = c( 'control','control','control','treatment','treatment','treatment'),
                       stringsAsFactors = FALSE)
mBAMs <- data.frame( bam = sub("_[012]","",targets$bam[c(1,4)]),
                     condition = c("control","treatment"))

gbcounts <- gbCounts(features=features, targets=targets,
                     minReadLength = 100, maxISize = 50000)
gbcounts

asd <- jCounts(counts=gbcounts, features=features, minReadLength=100)
asd
gb  <- gbDUreport(gbcounts, contrast = c(-1,1))
gb
jdur <- jDUreport(asd, contrast=c(-1,1))
jdur

sr <- splicingReport(gb, jdur, counts=gbcounts)
is <- integrateSignals(sr,asd)
exportIntegratedSignals(is,sr=sr,
                          output.dir = "example",
                          counts=gbcounts,features=features,asd=asd,
                          mergedBams = mBAMs)

```

