---
title: "An√°lisis de datos de RNA-Seq Parte 3 Alternative Splicing"
author: "Estefania Mancini"
date: "Marzo 2022"
output:
    html_document:
    toc: true # table of content true
    toc_depth: 3  # upto three depths of headings (specified by #, ## and ###)
    number_sections: true  ## if you want number sections at each table header
    toc_float: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Utilizaremos los siguientes  paquetes

* [edgeR](http://bioconductor.org/packages/release/bioc/html/edgeR.html)

* [ggplot2]

* [pheatmap]

* ASpli

* DEGSeq 
* tximport

* DRIMSeq 

* DEXSeq

* rnaseqDTU
# Que mas podemos hacer?

## Analisis de Splicing Alternativo

## Teoria 1 h

## Practica 1h30

## Isoformas (Salmon, practica)

[SALMON] (http://bioconductor.org/packages/release/workflows/html/rnaseqDTU.html)

RNA-seq workflow for differential transcript usage (DTU) following Salmon quantification. This workflow uses Bioconductor packages tximport, DRIMSeq, and DEXSeq to perform a DTU analysis on simulated data. It also shows how to use stageR to perform two-stage testing of DTU, a statistical framework to screen at the gene level and then confirm which transcripts within the significant genes show evidence of DTU.

### DTU: Differential Transcript Usage

https://f1000research.com/articles/7-952/v3

## Eventos (ASpli, practica)

