---
title: "An√°lisis de datos de RNA-Seq - Splicing Alternativo"
subtitle: "BioScienceApps"
author: "Estefania Mancini"
date: "18-21 de abril de 2022"
output:
    html_document:
    toc: true # table of content true
    toc_depth: 3  # upto three depths of headings (specified by #, ## and ###)
    toc_float: true
    code_folding: hide
    fig_caption: yes
    fig_height: 6
    fig_width: 6
    number_sections: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```


# ASpli

```{r design, echo=TRUE, eval=TRUE}
library(GenomicFeatures)

TxDb <- makeTxDbFromGFF(
  file="~/Documents/BioApps/ssh/chr14.gtf",
  format="gtf")

library(ASpli)
features <- binGenome(TxDb) 

library(RNAseqData.HNRNPC.bam.chr14)
genome <- TxDb
targets <- data.frame(bam=RNAseqData.HNRNPC.bam.chr14_BAMFILES,
                      condition=c(rep("CT",4),rep("KD",4)))

gbcounts <- gbCounts(features=features, targets=targets,
                     minReadLength = 100, maxISize = 50000)
asd <- jCounts(counts=gbcounts, features=features, minReadLength=100)
gb <-  gbDUreport(gbcounts, contrast = c(-1,1))
jdur <- jDUreport(asd, contrast=c(-1,1), strongFilter = FALSE)
jdur

###########################
mBAMs <- data.frame( bam = sub("_[012]","",targets$bam[c(1,4)]),
                       condition = c("CT","KD"))
mBAMs
###########################
sr <- splicingReport(gb, jdur, counts=gbcounts)
is <- integrateSignals(sr,asd)
exportIntegratedSignals(is,sr=sr,
                          output.dir = "aspliExample",
                          counts=gbcounts,features=features,asd=asd,
                          mergedBams = mBAMs)

```

